src/inspect_ai/_eval/score.py

```diff
@@ -284,30 +285,39 @@ async def run_score_task(
     # initialize active model and store
     init_task_context(model, model_roles)
     init_subtask_store(state.store)
-    transcript = Transcript()
-    init_transcript(transcript)
+    init_transcript(Transcript())
+
+    if state.scores is None:
+        state.scores = {}
+    existing_score_names = [*state.scores]
 
     results: dict[str, SampleScore] = {}
     async with span(name="scorers"):
         for scorer in scorers:
-            scorer_name = unique_scorer_name(scorer, list(results.keys()))
+            scorer_name = unique_scorer_name(
```

Comment:
Mostly reorganization to be consistent with the logic in `run.py` (which I intend to consolidate into a single code path soon)